app/models/job.rb

```diff
@@ -12,6 +12,11 @@ class Job < ActiveRecord::Base
   validate :validate_globally_unlocked
 
   ACTIVE_STATUSES = %w[pending running cancelling].freeze
+  VALID_STATUSES = ACTIVE_STATUSES + %w[failed errored succeeded cancelled].freeze
+
+  def self.valid_status?(status)
```

Comment:
might not even need this method ... `VALID_STATUSES.include?(status)` could live in the controller